Buildings/Fluid/HeatExchangers/BaseClasses/WetCoilUARated.mo

```diff
@@ -0,0 +1,192 @@
+  parameter MediumW.ThermodynamicState staWat=MediumW.setState_phX(
+    p=MediumW.p_default,h=MediumW.h_default,X=MediumW.X_default[1:MediumW.nXi]);
+
+  parameter Modelica.SIunits.SpecificHeatCapacity cpAir=
+    MediumA.specificHeatCapacityCp(staAir);
+  parameter Modelica.SIunits.SpecificHeatCapacity cpw=
+    MediumW.specificHeatCapacityCp(staWat);
+  parameter Modelica.SIunits.SpecificHeatCapacity cpEff(fixed=false, min= 0)
+    "Effective specific heat: change in saturated moist air enthalpy with respect to
+    temperature along the saturation line between inlet and outlet water temperatures";
+
+  constant Modelica.SIunits.SpecificEnthalpy hunit=1;
+  constant Modelica.SIunits.Temperature Tunit=1;
+  constant Modelica.SIunits.SpecificHeatCapacity cpunit=1;
+
+  Modelica.SIunits.AbsolutePressure pSatTWatIn=
+    Buildings.Utilities.Psychrometrics.Functions.saturationPressure(TWatIn)
+    "Saturation pressure of water at the water inlet temperature";
+  Modelica.SIunits.MassFraction X_wSatTWatIn=
+    Buildings.Utilities.Psychrometrics.Functions.X_pSatpphi(
+      pSat=pSatTWatIn, p=MediumA.p_default, phi=1)
+    "Mass fraction of water in saturated moist air at the water inlet temperature";
+  Modelica.SIunits.SpecificEnthalpy hSatTWatIn=
+    Buildings.Media.Air.specificEnthalpy_pTX(
+      p=MediumA.p_default, T=TWatIn, X={X_wSatTWatIn,1-X_wSatTWatIn})
+    "Enthalpy of saturated moist air at the water inlet temperature";
+
+  Modelica.SIunits.AbsolutePressure pSatTWatOut=
+    Buildings.Utilities.Psychrometrics.Functions.saturationPressure(TWatOut)
+    "Saturation pressure of water at the water oulet temperature";
+  Modelica.SIunits.MassFraction X_wSatTWatOut=
+    Buildings.Utilities.Psychrometrics.Functions.X_pSatpphi(
+      pSat=pSatTWatOut, p=MediumA.p_default, phi=1)
+    "Mass fraction of water in saturated moist air at the water outlet temperature";
+  Modelica.SIunits.SpecificEnthalpy hSatTWatOut=
+    Buildings.Media.Air.specificEnthalpy_pTX(
+      p=MediumA.p_default, T=TWatOut, X={X_wSatTWatOut,1-X_wSatTWatOut})
+    "Enthalpy of saturated moist air at the water oulet temperature";
+
+  parameter Modelica.SIunits.SpecificEnthalpy LMED(fixed=false)
+    "Log mean enthalpy difference";
+
+  parameter Modelica.SIunits.MassFlowRate UASta(fixed=false, min=0)
+    "Overall heat transfer coefficient for enthalpy difference";
+  parameter Modelica.SIunits.HeatFlowRate QTot_flow=mWat_flow*cpw*(TWatOut-TWatIn)
+    "Total heat flow from air to water stream";
+
+  parameter Modelica.SIunits.ThermalConductance UAAir(min=0,start=10,fixed=false)
+  "Air side convective heat transfer coefficient, including fin resistance";
+  parameter Modelica.SIunits.ThermalConductance UAWat(min=0,start=20,fixed=false)
+  "Water side convective heat transfer coefficient";
+
+  parameter Boolean IsFulDry(fixed=false);
+  parameter Boolean IsFulWet(fixed=false);
+  parameter Boolean IsParWet(fixed=false);
```

Comment:
All parameters, constants and variables must have a documentation string, and use lower case for variable names, e.g., `isFulDry`

Reply:
Resolved.